elkodon/src/sample_mut.rs

```diff
@@ -44,7 +45,7 @@ pub struct SampleMut<
     MessageType: Debug,
 > {
     publisher: &'publisher Publisher<'a, 'config, Service, MessageType>,
-    ptr: NonNull<MaybeUninit<Message<Header, MessageType>>>,
+    ptr: RawSampleMut<Header, MessageType>,
```

Comment:
Here `MessageType` can be either the actual `MessageType` or a `MaybeUninit<MessageType>`. I'm not quite sure how to name the generic parameter to make this visible to the user. Any suggestions?

Reply:
Is it though really important to the user? They will never create a sample directly, but they will call `loan` or `loan_uninit` which takes care of this.

By a comment in the documentation would be nice. And maybe illustrate this with a little example for the case where it is not `MaybeUninit<MessageType>` (the case where you loan and default construct)

Reply:
No, it is not important for the user but might confuse the developer when working on this code.